WordPress/Classes/ViewRelated/Comments/CommentsViewController.m

```diff
@@ -346,17 +347,33 @@ - (NSManagedObjectContext *)managedObjectContext
 
 - (NSFetchRequest *)fetchRequest
 {
-    NSFetchRequest *fetchRequest            = [NSFetchRequest fetchRequestWithEntityName:[self entityName]];
-    fetchRequest.predicate                  = [NSPredicate predicateWithFormat:@"(blog == %@ AND status != %@)", self.blog, CommentStatusSpam];
-    
-    NSSortDescriptor *sortDescriptorStatus  = [NSSortDescriptor sortDescriptorWithKey:@"status" ascending:NO];
-    NSSortDescriptor *sortDescriptorDate    = [NSSortDescriptor sortDescriptorWithKey:@"dateCreated" ascending:NO];
-    fetchRequest.sortDescriptors            = @[sortDescriptorStatus, sortDescriptorDate];
-    fetchRequest.fetchBatchSize             = CommentsFetchBatchSize;
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:[self entityName]];
+
+    if ([Feature enabled:FeatureFlagCommentFilters]) {
+        fetchRequest.predicate = self.filteredCommentsPredicate;
+    } else {
+        fetchRequest.predicate = [NSPredicate predicateWithFormat:@"(blog == %@ AND status != %@)", self.blog, CommentStatusSpam];
+    }
+
+    NSSortDescriptor *sortDescriptorStatus = [NSSortDescriptor sortDescriptorWithKey:@"status" ascending:NO];
+    NSSortDescriptor *sortDescriptorDate = [NSSortDescriptor sortDescriptorWithKey:@"dateCreated" ascending:NO];
+    fetchRequest.sortDescriptors = @[sortDescriptorStatus, sortDescriptorDate];
+    fetchRequest.fetchBatchSize = CommentsFetchBatchSize;
     
     return fetchRequest;
 }
 
+- (NSPredicate *)filteredCommentsPredicate
```

Comment:
It doesn't look like this is called when changing filters. It looks like the intent was to update the fetchRequest's predicate for the selected filter, however the fetch request's predicate isn't being updated and the fetched objects are not being refreshed. 
Try disabling wifi and testing and you'll see the list never updates when there is no internet connection.
The filtering appears to be happening in the service where comments that do not belong to the current filter are purged. (I dig this approach actually.)